main/ejml-ddense/src/org/ejml/dense/row/mult/MatrixVectorMult_DDRM.java

```diff
@@ -346,4 +348,85 @@ public static double innerProduct( double a[], int offsetA,
 
         return output;
     }
+
+    /**
+     * scalar = A<sup>T</sup>*B*C
+     *
+     * @param A (Input) vector
+     * @param B (Input) Matrix
+     * @param C (Input) vector
+     */
+    public static double innerProduct(DMatrixSparse A, DMatrix1Row B, DMatrixSparse C) {
+        if (A.getNumElements() < B.numRows)
```

Comment:
Can you make sure the inputs are vectors and throw an exception if they are not? Also shouldn't it be 'A.getNumElements() != B.numRows'?

If you rebase off of SNAPSHOT there's a new function MatrixFeatures.isVector() that should simplify the code a little bit